### PR TITLE
Update to readme for workflow

### PR DESCRIPTION
**Pull request by netskink**
@MikeFultonDev these are my notes regarding our work today.  There are a few points of clarification in the notes, but perhaps this will be a good starting point for discussion.

Probably should add something along these lines for creating the pull request yadda yadda

```

After commit, fork the meta repo in github.  

After that do this 

```
$ git add README.md
$ git branch johndev
$ git checkout johndev
$ get remote add mine git@github.com:netskink/meta.git
$ git push --set-upstream mine johndev
```

At this point, git will specify how to create a git pull request on meta via the push.

### CODE REVIEW

File: README.md
Comment (MikeFultonDev) [inline README.md:85]: change to $HOME/zopen/dev/meta

File: README.md
Comment (MikeFultonDev) [inline README.md:101]: If you wanted to enhance/improve zotsampleport - likely because you made changes to zopen-build or because you want to test zopen-build better, then you would:
```
cd $HOME/zopen/dev
git clone git@github.com:ZOSOpenTools/zotsampleport.git
cd zotsampleport
$ git branch mymod
$ git checkout mymod

<make your changes>
git add .
git commit -m 'comment'
git push --set-upstream origin mymod
```

You now need to be able to test zotsampleport before you merge your branch:
```
cd $HOME/zopen/dev/metaport
export ZOPEN_ZOT_SAMPLEPORT_BRANCH=mymod
zopen build
```
This will now use your branch `mymod` that you pushed to the server when running `zopen_check_build` testcase in the `tests` directory, by using your branch `mymod` instead of `main` when cloning `zotsampleport`

Comment (MikeFultonDev) [reply README.md:101]: Since this is specific to enhancing the testing of zopen-build, it should perhaps go after the other more general description?

**Review by MikeFultonDev — COMMENTED** [review]
thank you for expanding the docs - please ensure the original readme information also is incorporated - i think most of it is here.

File: README.md
Comment (netskink) [reply README.md:85]: You are welcome.   Yes, indeed, I'll try to merge the changes into the original so it flow better.

File: README.md
Comment (netskink) [reply README.md:101]: Sounds good.  I'll revisit this patch tomorrow.  I'll use your suggestions.  Thanks for all the good info.  You make this fun.

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:38]: Can you capitalize the first letter in the headers?

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:40]: I'd be more specific here and say: If you want to contribute to the meta repository, follow the steps below

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:68]: Are you able to use a different name for the branch? Something like mybranch or `devbranch`

File: README.md
Comment (netskink) [reply README.md:38]: Yes that would be wise

File: README.md
Comment (netskink) [reply README.md:40]: Yes, indeed. Thanks

File: README.md
Comment (netskink) [reply README.md:68]: Yes.

**Comment by netskink** [comment]
hmm. I just noticed this workflow has an error in this section:

Its a common practive to work in the `~/zopen/dev/` directory.

```
$ cd ~/zopen/dev
$ git clone git@github.com:ZOSOpenTools/meta.git
$ git clone git@github.com:ZOSOpenTools/metaport.git
```

I can't push to new branch for these repo's.  I need to push to my forked version.

```
$ git push --set-upstream origin findutilsportE
Enter passphrase for key '/z/jd895801/.ssh/id_ed25519':
ERROR: Permission to ZOSOpenTools/meta.git denied to netskink.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by MikeFultonDev** [comment]
@IgorTodorovskiIBM i will leave this to you because a) you aren’t on vacation and b) you know for better than me

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:48]: `~/zopen/usr/local/bin` is a configurable location. Perhaps you can update it as such:

Step 1:
Update your z/OS Open Tools meta package before proceeding

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:57]:
```suggestion
This clones the repos for development work.  Note this workflow
```

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:58]: This should not be a requirement

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:65]:
```suggestion
It is common practice to work in the `$HOME/zopen/dev/` directory.
```

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:68]: To make it more clear that ~ represents HOME
```suggestion
$ cd $HOME/zopen/dev
```

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:80]:
```suggestion
$ cd $HOME/zopen/dev/meta
```

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:88]:
```suggestion
$ cd $HOME/zopen/dev/meta
```

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:107]:
```suggestion
export ZOPEN_META_DEV_ROOT=${HOME}/zopen/dev/meta
```

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:112]:
```suggestion
Source the script to have environment variables set in your current shell.
```

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:115]:
```suggestion
$ cd $HOME/zopen/dev
```

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:144]: Join with previous sentence

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:150]: dupe command?

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:161]:
```suggestion
### Step 6 - Use the developer version of meta
```

**Comment by IgorTodorovskiIBM** [comment]
Thanks @netskink , added a few more comments, but looking good overall. Thanks for the doing this.

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:44]: Should this be $HOME/zopen?

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:118]: type on disable

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:141]: branchin -> branch in

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:135]: I would add a little blurb here saying that zopen build builds your meta branch and also validates it via its test cases

**Review by IgorTodorovskiIBM — APPROVED** [review]
LGTM, other than a few minor comments

**Comment by netskink** [comment]
> /run tests

Is that something for me or some sort of trigger for github actions?

**Comment by IgorTodorovskiIBM** [comment]
> > /run tests
> 
> Is that something for me or some sort of trigger for github actions?

Yes, it triggers the github action. 

I'm seeing this issue:
```
Installing package: zotsampleport
***ERROR: Could not get all releases for zotsampleport
Test of identifying port suffix failed.
/jenkins/PortBuild/meta/tests/zopen_check_install 43: FSUM7325 return: not executing function
FAIL: /jenkins/PortBuild/meta/tests/zopen_check_install
```

I see, so https://github.com/ZOSOpenTools/meta/pull/575 needs to be merged in first. We currently don't have a way to test anything other than the main brain via github actions.

File: README.md
Comment (IgorTodorovskiIBM) [inline README.md:44]:
```suggestion
your installation of zopen is installed in `$HOME/zopen`.
```

File: tests/zopen_check_abanner
Comment (IgorTodorovskiIBM) [inline tests/zopen_check_abanner:10]: I wonder if this should be moved to https://github.com/ZOSOpenTools/metaport/blob/main/buildenv#L79?

Comment (netskink) [reply tests/zopen_check_abanner:10]: Got me.  there are two banners. One is the generic which shows the start.  The other has a specific name for a specific test.   They could be included or not based upon usage.  ie, you want to see when tests begin, use the base one.  Want to see a specific test?  Use the second variant with the test name in the banner text.

Comment (netskink) [reply tests/zopen_check_abanner:10]: That makes sense.  I’ll try to refit.  Many thanks!

File: tests/include/banner.sh
Comment (IgorTodorovskiIBM) [inline tests/include/banner.sh:1]: Can you remove extra newlines here?

Do you plan to add the banner at the start in zopen_check()?

Comment (netskink) [reply tests/include/banner.sh:1]: Yes I can remove them. I will do so.  I can add to beginning as well.  My apologies I thought you just wanted to pull them in individually.

**Pull request closed without merging**
